src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,165 @@
+  //     const X509SecurityReq = securityRequirements.find(sec => sec.type() === 'X509')
+  //     const url = new URL(this.AsyncAPIServer.url())
+
+  //     const certsConfig = process.env.GLEE_SERVER_CERTS?.split(',').map(t => t.split(':'))
+  //     const certs = certsConfig?.filter(tuple => tuple[0] === this.serverName)?.map(t => fs.readFileSync(t[1])) // eslint-disable-line security/detect-non-literal-fs-filename
+
+  //     this.kafkaConsumer = Kafka.connect({
+  //       host: url.host,
+  //       port: url.port || (url.protocol === 'kafka:' ? 1883 : 8883),
+  //       protocol: url.protocol.substr(0, url.protocol.length - 1),
+  //       clientId: serverBinding && serverBinding.clientId,
+  //       clean: serverBinding && serverBinding.cleanSession,
+  //       will: serverBinding && serverBinding.will && {
+  //         topic: serverBinding && serverBinding.lastWill && serverBinding.lastWill.topic ? serverBinding.lastWill.topic : undefined,
+  //         qos: serverBinding && serverBinding.lastWill && serverBinding.lastWill.qos ? serverBinding.lastWill.qos : undefined,
+  //         payload: serverBinding && serverBinding.lastWill && serverBinding.lastWill.message ? serverBinding.lastWill.message : undefined,
+  //         retain: serverBinding && serverBinding.lastWill && serverBinding.lastWill.retain ? serverBinding.lastWill.retain : undefined,
+  //       },
+  //       keepalive: serverBinding && serverBinding.keepAlive,
+  //       username: userAndPasswordSecurityReq ? process.env.GLEE_USERNAME : undefined,
+  //       password: userAndPasswordSecurityReq ? process.env.GLEE_PASSWORD : undefined,
+  //       ca: X509SecurityReq ? certs : undefined,
+  //     })
+
+  //     this.kafka.on('connect', () => {
+        
+
+  //       if (Array.isArray(subscribedChannels)) {
+  //         subscribedChannels.forEach((topic) => {
+  //           const operation = this.parsedAsyncAPI.channel(channel).publish()
+  //           const binding = operation.binding('mqtt')
+  //           this.kafka.subscribe(channel, {
+  //             qos: binding && binding.qos ? binding.qos : 0,
+  //           })
+  //         })
+  //       }
+
+  //       resolve(this)
+  //     })
+
+  //     this.kafka.on('message', (topic, partition, message) => {
+  //       const msg = this._createMessage(message as IPublishPacket)
+  //       this.emit('message', msg, this.kafka)
+  //     })
+
+  //     this.kafka.on('reconnect', () => {
+  //       this.emit('reconnect', {
+  //         connection: this.kafka,
+  //         channels: this.channelNames,
+  //       })
+  //     })
+      
+  //     this.kafka.on('close', () => {
+  //       this.emit('close', {
+  //         connection: this.kafka,
+  //         channels: this.channelNames,
+  //       })
+  //     })
+
+  //     this.kafka.on('error', (error) => {
+  //       this.emit('error', error)
+  //     })
+  // }
+
+  _send(message: GleeMessage): Promise<void> {
```

Comment:
@Ruchip16 do you need help implementing this part of the binding?

Reply:
I am currently implementing that as per my understanding, I would push the changes once I am done and then we can have a meeting to discuss them, wyt?

Reply:
Absolutely 👍 We can also discuss them here before the meeting, so everyone is involved in the discussion.

Reply:
donee